src/hooks/Auth.ts

```diff
@@ -105,15 +106,25 @@ export const useAuth = ({
         return;
       }
       if (configuration) {
-        await performRefreshTokenRequest(
-          configuration,
-          options.clientId,
-          options.redirectUrl,
-          savedRefreshToken,
-          options.tokenRequest?.extras,
-        )
-          .then(setTokenResponse)
-          .catch(console.error);
+        try {
+          const response = await performRefreshTokenRequest(
+            configuration,
+            options.clientId,
+            options.redirectUrl,
+            savedRefreshToken,
+            options.tokenRequest?.extras,
+          );
+          setTokenResponse(response);
+        } catch (err) {
+          if (err instanceof AppAuthError) {
+            const statusCode = Number.parseInt(err.message, 10);
+            if (statusCode >= 400 && statusCode < 500) {
+              // HTTP client error -> token is probably expired
+              await storage.removeItem(AUTH_REFRESH_TOKEN_KEY);
+            }
```

Comment:
shouldn't we still handle all other errors (with console.error)?

Reply:
Fixed in 9f881de